### ISSUE TITLE
Addon not working for me, help much appreciated.

### ISSUE THREAD

**Issue by Coonaps** [ogri-la/wowman#47]
https://imgur.com/a/lpx0A70

Configured the AddOns directory but getting error messages.

**Comment by torkus** [comment]
I can definitely see an error is happening there!

Could you do me a favour and start wowman from a terminal? If you're running wowman directly with java, then it would look like this:

`java -jar wowman-x.x.x-standalone.jar --verbosity debug`

If you're running Arch and installed it from the AUR, then it would look like this:

`wowman --verbosity debug`

You should then get a list of output in the terminal, including the stacktrace I need to help fix this problem for you.

**Comment by torkus** [comment]
Also, please ensure you're running the latest version of wowman.

**Comment by layday** [comment]
I encountered the same issue after upgrading to 0.9.0 though I wasn't sure if I should report it because I don't use Linux. This happens when you open the add-on folder in the picker, then press 'Select' without a folder being selected (highlighted). wowman (or the underlying library) expects that you should highlight the add-on folder, then press select _without_ opening it.  Here's the full trace:

```
clojure.lang.ExceptionInfo: Call to wowman.core/set-addon-dir! did not conform to spec:
                            "[...]/World of Warcraft/_classic_/Interface/AddOns/AddOns" - failed: directory? in: [0] at: [:args :addon-dir] spec: :wowman.specs/extant-dir

        clojure.spec.alpha/args: ("[...]/World of Warcraft/_classic_/Interface/AddOns/AddOns")
    clojure.spec.alpha/failure: :instrument
    clojure.spec.alpha/problems: [{:path [:args :addon-dir],
                                :pred me.raynes.fs/directory?,
                                :val
                                "[...]/World of Warcraft/_classic_/Interface/AddOns/AddOns",
                                :via [:wowman.specs/extant-dir :wowman.specs/extant-dir],
                                :in [0]}]
        clojure.spec.alpha/spec: #object[clojure.spec.alpha$regex_spec_impl$reify__2509 0x4e8832ee "clojure.spec.alpha$regex_spec_impl$reify__2509@4e8832ee"]
    clojure.spec.alpha/value: ("[...]/World of Warcraft/_classic_/Interface/AddOns/AddOns")
    orchestra.spec.test/caller: {:file "gui.clj",
                                :line 239,
                                :var-scope wowman.ui.gui/configure-app-panel,
                                :local-fn picker}
```

This works:

<img width="662" alt="Screenshot 2019-09-18 at 05 27 26" src="https://user-images.githubusercontent.com/31134424/65102886-327fab80-d9d5-11e9-9b46-3b2f7576121d.png">

This doesn't:

<img width="662" alt="Screenshot 2019-09-18 at 05 28 32" src="https://user-images.githubusercontent.com/31134424/65102899-37dcf600-d9d5-11e9-8c2a-8fa3d8aac214.png">

**Comment by layday** [comment]
Then again, uh, I should've probably reported this separately.

**Comment by torkus** [comment]
@layday it's fine.

@Coonaps , do you still have a problem if you actually select the folder (rather than just 'open' it)?

**Comment by torkus** [comment]
(just for posterity)

![qkp0LGn](https://user-images.githubusercontent.com/48790775/65154443-b172f380-da6a-11e9-91c7-0f0cd7c697b1.png)

**Comment by Coonaps** [comment]
Ran the program in debug mode, here is the result, still not working for me

![Screenshot from 2019-09-18 16:11:41](https://user-images.githubusercontent.com/55458327/65156408-5a036200-da2f-11e9-8195-8a84c7d9bde1.png)

**Comment by torkus** [comment]
that's really helpful, thank you.

My gut feeling is that you have an addon with a malformed `.toc` file in it. I may need to ask for more information about which addons you have installed, but let me debug what you've given me first.

**Comment by Coonaps** [comment]
Sure anytime, I am grateful for your efforts and hope we get things working.

> that's really helpful, thank you.
> 
> My gut feeling is that you have an addon with a malformed `.toc` file in it. I may need to ask for more information about which addons you have installed, but let me debug what you've given me first.

**Comment by torkus** [comment]
hey @Coonaps , could you try using this patched version of wowman and letting me know how you go? You'll need to rename the file from `.jar.zip` to just `.jar` (Github doesn't allow `.jar` files to be uploaded).

[wowman-0.9.0-patched.jar.zip](https://github.com/ogri-la/wowman/files/3630121/wowman-0.9.0-patched.jar.zip)
sha256: 8103cf0dc0699a8d13c3a5f0df8873fe9fafc396499dfaffba76d7efe8199811

[PR for this patched version](https://github.com/ogri-la/wowman/pull/51)

The problem is that one of the addons you have installed is missing an important value (it's `Title`) in it's metadata that should be present for *all* addons. This change ensures that a failure to parse an addon's metadata won't prevent wowman from running as well as substituting the missing value with something similar that will have to make do. 

It will issue a warning telling you which adding is missing this value.

**Comment by Coonaps** [comment]
You are a damn genius! Thumbs up mate! Attached the screenshot for you. Are you gonna update the "real" version here?

![Screenshot from 2019-09-20 12:43:56](https://user-images.githubusercontent.com/55458327/65321427-75dc4480-dba4-11e9-9c15-2be0c510b3ff.png)

**Comment by torkus** [comment]
ah good to see, I'm glad it's working for you

> Are you gonna update the "real" version here?

I'm preparing a 0.9.1 release here: https://github.com/ogri-la/wowman/pull/54

It contains two other small fixes, including the one @layday appended to this ticket originally

**Comment by torkus** [comment]
also, I'm not sure if you've read the README, but your next step now that it's working for you is to use the 're-install all' feature under the 'Addons' menu.

It will download the latest versions of the addons that match in the catalog and you'll find some of those that don't match will disappear, as these are sub-modules of the main addon and, as soon as we can tie them to the main addon, we don't need to see them anymore

**Comment by torkus** [comment]
0.9.1 has been released, closing issue.

all the best, @Coonaps , let me know if you get any more errors

**Comment by Coonaps** [comment]
re-install all gives me the following error 

![Screenshot from 2019-09-20 13:57:27](https://user-images.githubusercontent.com/55458327/65325284-b17c0c00-dbae-11e9-859d-8419a8c5e4bf.png)

**Comment by torkus** [comment]
I can see something is wrong but I don't know what, so screenshots are not good enough anymore. 

I need you to scroll up in that terminal and select everything from when you started the app to the very last thing. The output may be too large to paste directly into github, so try pasting it into a text file first and then attaching the text file to the bug. The more information you give me, the less guessing I have to do.

**Comment by torkus** [comment]
I just discovered this bug while attempting to replicate yours: https://github.com/ogri-la/wowman/issues/55

it may be related.

if you're sick of this bugginess, I can appreciate that. A more stable version of wowman was [0.8.2](https://github.com/ogri-la/wowman/releases/tag/0.8.2), although that release did not have multiple directory support or switching between retail and classic versions.

**Comment by torkus** [comment]
I think the problem is that you a classic-only addon from Curseforge (AtlasLootClassic) and are trying to install it into a directory that is 'retail'. Because Curseforge allows multiple different 'releases' for the same addon, I use the selection of that dropdown, either 'classic' or 'retail', to select which release of the addon to install. In this case there is no retail version and you get those errors.

It's not a quick fix this time and I'll need to think about how to best tackle it.

I appreciate you taking the time to report these problems, @Coonaps .

**Comment by torkus** [comment]
hey hey! I'm able to replicate your problem. It's different from the one discovered above. Investigating ...

**Comment by torkus** [comment]
fix is available in 0.9.2 

please let me know how it goes for you

**Comment by Coonaps** [comment]
Working now, thank you!
